### PR TITLE
overworld | quest list UI tracks solveable and solved quests more clearly

### LINKED ISSUE #362: quest solved indicator, improve quest list

### PR DESCRIPTION
**Pull request by mango-gremlin**
## Motivation
closes #362 

To make it more clear you can solve a quest and when you solved it.

## What was changed/added
Instead of having a quest list with current quests where finished quests just disappear, now:
- new quests are added to quest list with empty checkmark
- when quest is solveable, a speech bubble with a ! appears next to the quest
- when solved, this speech bubble disappears and instead the quest is checked off, but not removed from the ui

## Testing
Everything works however, i ran into 1 problem:
- you havent talked to the npc at all, but you have the item that solves the quest
- when you talk to npc, the quest is solved, you get the rewards, but the ui doesnt add the quest, doesnt mark it off either :(

everything else works:

https://github.com/mango-gremlin/arch-enemies/assets/104830903/22b152ec-569a-49e4-a86c-da0dc9616ad8

What doesnt work:

https://github.com/mango-gremlin/arch-enemies/assets/104830903/f2423e03-d8a3-41e5-983e-53545fde9810

**I assume this has something to do with how I add quests to the list, or when? If anyone has an idea please share...**

### CODE REVIEW

File: Arch-enemies/overworld/ui/overworld_ui/overworld_ui.gd [hunk 1]
```diff
@@ -40,14 +49,20 @@ var player_item_inventory:Dictionary = {}:
 		player_item_inventory = new_dict
 		_update_item_inventory_label()
 
-var player_animal_inventory:Dictionary = {}:
+var player_animal_inventory:Dictionary = {}#:
```
Comment (mango-gremlin) [inline Arch-enemies/overworld/ui/overworld_ui/overworld_ui.gd:52]: i just realised i commented this on accident. will fix it asap, sorry!

**Comment by Kabooni** [comment]
Lgtm, besides the bug it worked like intended. If needed, I approve it once the bug is fixed.

**Comment by ScatteredDrifter** [comment]
@mango-gremlin @Kabooni maybe have a look again /

**Comment by mango-gremlin** [comment]
@ScatteredDrifter 's fix works! thanks!
I also added saving quests to the singleton when leaving scene and loading when coming back.

**Review by Kabooni — APPROVED** [review]
There's a known bug with solved quests disappearing after changing scenes, but it was decided to look into it later.

**Review by marcel-iv — APPROVED** [review]
as noted elsewhere, a solved quest disappears when dialogue is **not** skipped, and re-entering the main scene. 

Otherwise, works fine, the bug will be tracked in another issue

**Pull request merged**